### ISSUE TITLE
Release an improved version of OONI Run to better meet community needs

### ISSUE THREAD

**Issue by gabelula** [ooni/probe#814]
We will implement a new OONI Run link format and roll it out in production.

**Comment by lorenzoPrimi** [comment]
@hellais do you have any specification?

**Comment by hellais** [comment]
I did some research into this here: https://gist.github.com/hellais/c7b041f039e70d40bb0954eb297dad03

I think it would also be useful to prior to rolling out a new link format gather a bit more input from users that have been experiencing issue with OONI Run links and maybe do some focussed user research (or just a survey?).

OONI Run has been a big source of confusion for many of our users and we have had them complain about it being "broken" or "not working" for some time, yet it's not super clear to me yet, what exactly are the main sources of the issues.

cc @agrabeli @holantonela @elioqoshi

**Comment by elioqoshi** [comment]
I think it would be a great first step to start with a survey and later on do some user research with a more curated list of people.

I'd also like to understand better how well users of OONI Probe are aware of OONI Run, especially after the 2.0 release. I strongly suspect we don't explain OONI Run properly in the whole OONI ecosystem, especially OONI Probe (something which we lengthily discussed also over Adarsh's last User Testing results).

**Comment by hellais** [comment]
I think we need to first do some research to understand the limitations of our current approach and improve upon it only once we have those clear.

**Comment by hellais** [comment]
Another idea to include in the new version of OONI Run is to possibly use this channel to also encode censorship circumvention related settings such as alternative bouncer/collector addresses as suggested by @andresazp.

**Comment by bassosimone** [comment]
Yes, this seems an excellent idea!

**Comment by hellais** [comment]
Another issue that keeps popping up and I think should be addressed by the new format is that "Internal server error" message. I saw this mentioned in the report from @lorenzoPrimi.

The root cause of this is that the path URI component is too long and is getting transmitted and triggerring an nginx error. A way around this (as outlined in [OONI Run Link research](https://gist.github.com/hellais/c7b041f039e70d40bb0954eb297dad03)) is to put the link in the fragment component (this is not sent on the wire).

We can fix the internal server error on our end by tweaking nginx, but it would still occur in the case of middleboxes failing to parse too long path components, so it's worth addressing at the OONI Run link format for sure.

**Comment by bassosimone** [comment]
Can you clarify "in the case of middleboxes failing to parse too long path components"? Are talking about middleboxes that intercept HTTPS traffic or are we using HTTP for OONI run?

**Comment by hellais** [comment]
@bassosimone I guess this would only apply to cases in which they are hitting the OONI Run link on HTTP, which could be the case if the link is clicked on via some social media app (I haven't actually confirmed this).

Perhaps this is not so much of a concern, but I figured I would mention it in this ticket as it's relevant to the new format design.

**Comment by bassosimone** [comment]
Makes sense.

**Comment by lorenzoPrimi** [comment]
That link that triggered "Internal server error" message wasn't really too long, it only included 3 urls

**Comment by hellais** [comment]
> That link that triggered "Internal server error" message wasn't really too long, it only included 3 urls

Can you share the link?

**Comment by andresazp** [comment]
People complain to me about that too, if they retry, same deal.

- AA

**Comment by hellais** [comment]
@andresazp it would be super useful if we could have a sample link that is triggering the error and maybe also a screenshot of the error from the phone + how they reached reached the link (clicked on it in an email, clicked on a link on twitter, etc.).

**Comment by lorenzoPrimi** [comment]
@hellais it's on the email I sent last weekend

**Comment by lorenzoPrimi** [comment]
The only way to open it is to use the option "Open in browser"

![whatsapp image 2019-02-17 at 14 01 55](https://user-images.githubusercontent.com/1616097/53075158-10957780-351f-11e9-82cc-9c5e2a80ec2e.jpeg)
 
![whatsapp image 2019-02-17 at 14 00 55](https://user-images.githubusercontent.com/1616097/53075170-15f2c200-351f-11e9-9ade-ff7c6a84f034.jpeg)

**Comment by andresazp** [comment]
Users som have complained about some ooni links like (working well for others): 

destination: "https://run.ooni.io/nettest?tn=web_connectivity&ta=%7B%22urls%22%3A%5B%22http%3A%2F%2Fwww.ultrasurf.us%2F%22%2C%22http%3A%2F%2Fanonymizer.ru%2F%22%2C%22http%3A%2F%2Fwww.tunnelbear.com%2F%22%2C%22https%3A%2F%2Ftwitter.com%2F%22%2C%22http%3A%2F%2Ftwitter.com%2F%22%2C%22http%3A%2F%2Fmobile.twitter.com%2F%22%2C%22https%3A%2F%2Fmobile.twitter.com%2F%22%2C%22https%3A%2F%2Fpbs.twimg.com%2F%22%2C%22https%3A%2F%2Fabs.twimg.com%2F%22%2C%22http%3A%2F%2Ffacebook.com%2F%22%2C%22http%3A%2F%2Fstatic.xx.fbcdn.net%2F%22%2C%22https%3A%2F%2Fwww.youtube.com%2F%22%2C%22http%3A%2F%2Fyoutube.com%2F%22%2C%22https%3A%2F%2Fssl.gstatic.com%2F%22%2C%22http%3A%2F%2Fajax.googleapis.com%2F%22%2C%22http%3A%2F%2Flh3.googleusercontent.com%2F%22%2C%22http%3A%2F%2Finstagram.com%2F%22%2C%22https%3A%2F%2Finstagram.com%2F%22%2C%22http%3A%2F%2Fapi.instagram.com%2F%22%2C%22https%3A%2F%2Fapi.instagram.com%2F%22%2C%22https%3A%2F%2Fwww.pscp.tv%2F%22%2C%22https%3A%2F%2Fapi.periscope.tv%2F%22%2C%22http%3A%2F%2Fgmail.com%2F%22%2C%22https%3A%2F%2Fmail.google.com%2F%22%2C%22https%3A%2F%2Fblogger.com%2F%22%2C%22https%3A%2F%2Fgetintra.org%2F%22%2C%22https%3A%2F%2Fgetoutline.org%2F%22%2C%22https%3A%2F%2Fgetlantern.org%2F%22%2C%22http%3A%2F%2Fcheck.googlezip.net%2F%22%2C%22http%3A%2F%2Fcheck.googlezip.net%2Fconnect%2F%22%2C%22http%3A%2F%2Fdatasaver.googleapis.com%2F%22%2C%22https%3A%2F%2Fdatasaver.googleapis.com%2F%22%2C%22https%3A%2F%2Faccounts.google.com%2F%22%2C%22http%3A%2F%2Fvoluntariosxvenezuela.com%2F%22%2C%22https%3A%2F%2Fvoluntariosxvenezuela.com%2F%22%2C%22http%3A%2F%2Fwww.voluntariosxvenezuela.com%2F%22%2C%22https%3A%2F%2Fwww.voluntariosxvenezuela.com%2F%22%2C%22http%3A%2F%2Fvoluntariovenezuela.com%2F%22%2C%22http%3A%2F%2Fwww.voluntariovenezuela.com%2F%22%2C%22https%3A%2F%2Fwww.voluntariovenezuela.com%2F%22%2C%22http%3A%2F%2Fvoluntariosporvenezuela.com%2F%22%2C%22https%3A%2F%2Fvoluntariosxporvenezuela.com%2F%22%2C%22https%3A%2F%2Fnordvpn.com%2F%22%2C%22https%3A%2F%2Fwww.tuneloso.com%2F%22%2C%22http%3A%2F%2Fwww.tuneloso.com%2F%22%2C%22https%3A%2F%2Fwww.opera.com%2F%22%2C%22https%3A%2F%2Fd2zfqthxsdq309.cloudfront.net%2F%22%2C%22http%3A%2F%2Ftor.eff.org%2F%22%2C%22http%3A%2F%2Fwww.anonymouse.org%2F%22%2C%22http%3A%2F%2Fanonym.to%2F%22%2C%22http%3A%2F%2Fwww.hola.org%2F%22%2C%22https%3A%2F%2Fguardianproject.info%2F%22%2C%22http%3A%2F%2Fwww.anonymizer.com%2F%22%2C%22http%3A%2F%2Fuproxy.org%2F%22%2C%22http%3A%2F%2Fwww.safervpn.com%2F%22%2C%22https%3A%2F%2Fhola.org%2F%22%2C%22http%3A%2F%2Fwww.airvpn.org%2F%22%2C%22http%3A%2F%2Fwww.3proxy.com%2F%22%2C%22https%3A%2F%2Fmeek-reflect.appspot.com%2F%22%2C%22http%3A%2F%2Fproxify.com%2F%22%2C%22https%3A%2F%2Fajax.aspnetcdn.com%2F%22%2C%22http%3A%2F%2Ftorproject.org%2F%22%2C%22https%3A%2F%2Fd0.awsstatic.com%2F%22%2C%22http%3A%2F%2Fwww.blockedsiteaccess.com%2F%22%2C%22https%3A%2F%2Fa0.awsstatic.com%2F%22%2C%22https%3A%2F%2Fbridges.torproject.org%2F%22%2C%22https%3A%2F%2Fa1.awsstatic.com%2F%22%2C%22https%3A%2F%2Fsecure.logmein.com%2F%22%2C%22http%3A%2F%2Fzenvpn.net%2F%22%2C%22http%3A%2F%2Fwww.purevpn.com%2F%22%2C%22https%3A%2F%2Fgeti2p.net%2F%22%2C%22https%3A%2F%2Fcdnjs.cloudflare.com%2F%22%2C%22http%3A%2F%2Fwww.luminati.io%2F%22%2C%22http%3A%2F%2Fwww.ultravpn.fr%2F%22%2C%22http%3A%2F%2Fhidemyass.com%2F%22%2C%22http%3A%2F%2Fanonymouse.ws%2Fanonwww.html%2F%22%2C%22https%3A%2F%2Fwww.tuneloso.com%2F%22%2C%22http%3A%2F%2Fwww.tuneloso.com%2F%22%2C%22http%3A%2F%2F2ip.ru%2F%22%2C%22http%3A%2F%2Fwww.swissvpn.net%2F%22%2C%22http%3A%2F%2Fdroidvpn.com%2F%22%2C%22http%3A%2F%2Fopenvpn.net%2F%22%2C%22http%3A%2F%2Fwww.hotspotshield.com%2F%22%2C%22https%3A%2F%2Fgetfoxyproxy.org%2F%22%2C%22http%3A%2F%2Fwww.vpn4all.com%2F%22%2C%22https%3A%2F%2Ftails.boum.org%2F%22%2C%22https%3A%2F%2Fgetoutline.org%2F%22%2C%22http%3A%2F%2Fwww.nordvpn.com%2F%22%2C%22http%3A%2F%2Fprivoxy.org%2F%22%2C%22https%3A%2F%2Fstrongvpn.com%2F%22%2C%22http%3A%2F%2Fwww.zenmate.com%2F%22%2C%22http%3A%2F%2Fanonymouse.org%2F%22%2C%22http%3A%2F%2Ftunnelbear.com%2F%22%2C%22http%3A%2F%2Fwww.mullvad.net%2F%22%2C%22http%3A%2F%2Fwww.disconnect.me%2F%22%2C%22http%3A%2F%2Fno-ip.org%2F%22%2C%22http%3A%2F%2Fwww.strongvpn.com%2F%22%2C%22http%3A%2F%2Fwww.cactusvpn.com%2F%22%2C%22https%3A%2F%2Fwww.opendns.com%2F%22%2C%22https%3A%2F%2Fpsiphon.ca%2F%22%2C%22http%3A%2F%2Fultrasurf.us%2F%22%5D%7D&mv=1.2.0

**Comment by hellais** [comment]
Thanks for these details. It's very useful.

@lorenzoPrimi are these screenshots you post related to using the twitter app?

Is the relevant tweet this one: https://twitter.com/thanittang/status/1097026214198296576 &  https://t.co/yXu3DxWtvq?

**Comment by lorenzoPrimi** [comment]
@hellais yes that is the android twitter app

**Comment by hellais** [comment]
We should schedule a meeting for January to discuss the next steps in how we are going to iterate upon this.

**Comment by hellais** [comment]
We have released OONI Run v2 🥳 

See: https://ooni.org/post/2024-launch-ooni-run-v2/
